Signal/src/ViewControllers/ConversationView/Cells/OWSCallMessageCell.m

```diff
@@ -0,0 +1,386 @@
+    self.vStackView.layoutMargins = UIEdgeInsetsMake(self.conversationStyle.textInsetTop,
+        self.conversationStyle.textInsetHorizontal,
+        self.conversationStyle.textInsetBottom,
+        self.conversationStyle.textInsetHorizontal);
+}
+
+- (BOOL)hasFooter
+{
+    return !self.viewItem.shouldHideFooter;
+}
+
+- (CGFloat)circleSize
+{
+    return 48.f;
+}
+
+- (UIColor *)textColorForCall:(TSCall *)call
+{
+    return [self.conversationStyle bubbleTextColorWithCall:call];
+}
+
+- (UIColor *)bubbleColorForCall:(TSCall *)call
+{
+    return [self.conversationStyle bubbleColorWithCall:call];
+}
+
+- (UIColor *)iconColorForCall:(TSCall *)call
+{
+    switch (call.callType) {
+        case RPRecentCallTypeIncoming:
+        case RPRecentCallTypeOutgoing:
+        case RPRecentCallTypeIncomingIncomplete:
+        case RPRecentCallTypeOutgoingIncomplete:
+            return [UIColor ows_greenColor];
+        case RPRecentCallTypeIncomingMissed:
+        case RPRecentCallTypeIncomingMissedBecauseOfChangedIdentity:
+        case RPRecentCallTypeIncomingDeclined:
+            return [UIColor ows_redColor];
+    }
+}
+
+- (UIImage *)iconForCall:(TSCall *)call
+{
+    UIImage *result = nil;
+    switch (call.callType) {
+        case RPRecentCallTypeIncoming:
+        case RPRecentCallTypeOutgoing:
+        case RPRecentCallTypeIncomingIncomplete:
+        case RPRecentCallTypeOutgoingIncomplete:
+            result = [UIImage imageNamed:@"phone-up"];
+            break;
+        case RPRecentCallTypeIncomingMissed:
+        case RPRecentCallTypeIncomingMissedBecauseOfChangedIdentity:
+        case RPRecentCallTypeIncomingDeclined:
+            result = [UIImage imageNamed:@"phone-down"];
+            break;
+    }
+    OWSAssert(result);
+    return result;
+}
+
+- (void)applyTitleForCall:(TSCall *)call label:(UILabel *)label
+{
+    OWSAssert(call);
+    OWSAssert(label);
+
+    [self configureFonts];
+
+    label.text = [self titleForCall:call];
+}
+
+- (NSString *)titleForCall:(TSCall *)call
+{
+    // We don't actually use the `transaction` but other sibling classes do.
+    switch (call.callType) {
+        case RPRecentCallTypeIncoming:
+        case RPRecentCallTypeOutgoing:
+        case RPRecentCallTypeOutgoingIncomplete:
+        case RPRecentCallTypeIncomingIncomplete:
+            return NSLocalizedString(@"CALL_DEFAULT_STATUS",
+                @"Message recorded in conversation history when local user is making or has completed a call.");
+        case RPRecentCallTypeIncomingMissed:
+        case RPRecentCallTypeIncomingMissedBecauseOfChangedIdentity:
+            return NSLocalizedString(
+                @"CALL_MISSED", @"Message recorded in conversation history when local user missed a call.");
+        case RPRecentCallTypeIncomingDeclined:
+            return NSLocalizedString(
+                @"CALL_DECLINED", @"Message recorded in conversation history when local user declined a call.");
```

Comment:
Not sure if the new copy is final, but I figured it would be a good time to distinguish the copy used in conversation view from the copy used in notifications, home view snippets, etc.

Reply:
👍